lib/services/firebase_service.dart

```diff
@@ -247,14 +249,127 @@ class FirebaseService extends IFirebaseService {
     return map;
   }
 
-  Future<void> addNotification(Map<String, dynamic> data) async {
+  Future<void> addNotification(
+      {@required String userId,
+      @required String formId,
+      @required String formName}) async {
+    var creation = _calculationService.formatDate(date: DateTime.now());
+    var anSubCollection = await getLatestAnSubCollection(userId: userId);
+    var patientState = anSubCollection['state'];
+    Map<String, dynamic> dataToAdd = {
+      'formName': formName,
+      'formId': formId,
+      'userId': userId,
+      'creation': creation,
+      'patientState': patientState,
+      'seen': false,
+      'patientSeen': false,
+    };
     await _firestore
         .collection('Notifications')
-        .add(data)
+        .add(dataToAdd)
         .then((value) =>
             print('Successfully added $value to Notificaitons Collection'))
         .catchError((e) {
       print('$e failed to add notification to Notifications Collection');
     });
   }
+
+  Future<List<QueryDocumentSnapshot>> getNotificationList() async {
+    var data;
+    var storedUserId = UserStore.getValueFromStore('storedUserId');
+    var anSubcollection = await getLatestAnSubCollection(userId: storedUserId);
+    var patientState = anSubcollection["state"];
+    var checkState = enumToString(PatientState.postOperationHome);
+    if (patientState == checkState) {
+      data = await _firestore
+          .collection('Notifications')
+          .where('userId', isEqualTo: storedUserId)
+          .where('seen', isEqualTo: true)
+          .get();
+    } else {
+      data = await _firestore
+          .collection('Notifications')
+          .where('userId', isEqualTo: storedUserId)
+          .get();
+    }
+    return data.docs;
+  }
+
+  Future<List<Map<String, dynamic>>> getNotifications() async {
+    var notiList = await this.getNotificationList();
+    var returnList = notiList.map((user) async {
+      var notiCollection =
+          await _firestore.collection("Notifications").doc(user.id).get();
+      var seen = notiCollection['seen'];
+      if (seen == false) {
+        seen = "ยังไม่ได้ดำเนินการ";
+      } else {
+        seen = "ดำเนินการแล้ว";
+      }
+      var patientState = notiCollection['patientState'];
+      var formName = notiCollection['formName'];
+      formName = formNameModel[formName];
+      var time = notiCollection['creation'];
+      var formTime =
+          DateTime.fromMicrosecondsSinceEpoch(time.microsecondsSinceEpoch);
+      var formDateToShow = DateFormat('dd/MM/yyyy').format(formTime);
+      var formTimeToShow = "${DateFormat.Hm().format(formTime)} น.";
+
+      var patientSeen = notiCollection['patientSeen'] ?? '-';
+      if (patientSeen == false) {
+        patientSeen = "ยังไม่ได้อ่าน";
+      } else {
+        patientSeen = "อ่านแล้ว";
+      }
```

Comment:
This one as well `notiCollection['patientState']`  is a boolean. If it is null by some reason it would be assigned to be string `-`. Then it will boooom because it can't be compared to boolean in the following condition

Reply:
Solved ka